osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/networking/DlmsPushNotificationDecoder.java

```diff
@@ -130,28 +302,80 @@ private DlmsPushNotification buildPushNotification() {
         }
     }
 
-    private void decodeObisCodeData(final ChannelBuffer buffer) {
+    private void decodeObisCodeData(final ChannelBuffer buffer, Protocol protocol, boolean alarmRegisterExpected) throws UnrecognizedMessageDataException {
 
         final byte[] logicalNameBytes = new byte[NUMBER_OF_BYTES_FOR_LOGICAL_NAME];
         buffer.readBytes(logicalNameBytes, 0, NUMBER_OF_BYTES_FOR_LOGICAL_NAME);
 
-        if (Arrays.equals(SMS_OBISCODE_BYTES, logicalNameBytes)) {
-            this.builder.withTriggerType(PUSH_SMS_TRIGGER);
-        } else if (Arrays.equals(CSD_OBISCODE_BYTES, logicalNameBytes)) {
-            LOGGER.warn("CSD Push notification not supported");
-            this.builder.withTriggerType(PUSH_CDS_TRIGGER);
-        } else if (Arrays.equals(SCHEDULER_OBISCODE_BYTES, logicalNameBytes)) {
-            LOGGER.warn("Scheduler Push notification not supported");
-            this.builder.withTriggerType(PUSH_SCHEDULER_TRIGGER);
-        } else {
-            LOGGER.warn("Unknown Push notification not supported. Unable to decode");
-            this.builder.withTriggerType("");
+        try {
+            if (!alarmRegisterExpected && isLogicalNameSmsTrigger(logicalNameBytes, protocol)) {
+                this.builder.withTriggerType(PUSH_SMS_TRIGGER);
+            } else if (!alarmRegisterExpected && isLogicalNameCsdTrigger(logicalNameBytes, protocol)) {
+                LOGGER.warn("CSD Push notification not supported");
+                this.builder.withTriggerType(PUSH_CSD_TRIGGER);
+            } else if (!alarmRegisterExpected && isLogicalNameSchedulerTrigger(logicalNameBytes, protocol)) {
+                LOGGER.warn("Scheduler Push notification not supported");
+                this.builder.withTriggerType(PUSH_SCHEDULER_TRIGGER);
+            } else if (alarmRegisterExpected && isLogicalNameAlarmTrigger(logicalNameBytes, protocol)) {
+                this.builder.withTriggerType(PUSH_ALARM_TRIGGER);
+            } else {
+                LOGGER.warn("Unknown Push notification not supported. Unable to decode");
+                this.builder.withTriggerType("");
+            }
+        } catch (ProtocolAdapterException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1166)